R/visualization/plotBestScatter.R

```diff
@@ -170,48 +155,149 @@ loadBestScores <- function(scores_files) {
             stringr::str_detect(Dataset, "Sim", negate = TRUE),
-            !is.na(`Overall Score`)
+            stringr::str_detect(Dataset, "3batches", negate = TRUE),
+            !is.na(`Batch Correction`),
+            !is.na(`Bio conservation`)
         ) %>%
-        dplyr::mutate(
-            MethodVersion = factor(MethodVersion, levels = best_methods)
+        dplyr::select(
+            Dataset, MethodVersion, MethodLabel, `Batch Correction`,
+            `Bio conservation`, Output
         )
 
     return(best_scores)
 }
+
+getBestKeys <- function(type = c("RNA", "ATAC")) {
+
+    type <- match.arg(type)
+
+    dataset_key <- list(
+        RNA  = c(
+            pancreas                       = "Pancreas",
+            lung_atlas                     = "Lung",
+            immune_cell_hum                = "Immune (human)",
+            immune_cell_hum_mou            = "Immune (human/mouse)",
+            mouse_brain                    = "Mouse brain",
+            simulations_1_1                = "Sim 1",
+            simulations_2                  = "Sim 2"
+        ),
+        ATAC = c(
+            mouse_brain_atac_genes_large   = "ATAC large (genes)",
+            mouse_brain_atac_peaks_large   = "ATAC large (peaks)",
+            mouse_brain_atac_windows_large = "ATAC large (windows)",
+            mouse_brain_atac_genes_small   = "ATAC small (genes)",
+            mouse_brain_atac_peaks_small   = "ATAC small (peaks)",
+            mouse_brain_atac_windows_small = "ATAC small (windows)"
+        )
+    )
+
+    best_methods <- list(
+        RNA = c(
+            "Scanorama_embed_HVG_scaled",
+            "scANVI*_embed_HVG_unscaled",
+            "scGen*_gene_HVG_unscaled",
+            "fastMNN_embed_HVG_unscaled",
+            "BBKNN_graph_HVG_unscaled",
+            "Scanorama_gene_HVG_scaled",
+            "scVI_embed_HVG_unscaled",
+            "Seurat v3 RPCA_gene_HVG_unscaled",
+            "Harmony_embed_HVG_unscaled",
+            "Seurat v3 CCA_gene_HVG_unscaled",
+            "fastMNN_gene_HVG_unscaled",
+            "Conos_graph_FULL_unscaled",
+            "ComBat_gene_HVG_unscaled",
+            "MNN_gene_HVG_unscaled",
+            "trVAE_embed_HVG_unscaled",
+            "DESC_embed_FULL_unscaled",
+            "LIGER_embed_HVG_unscaled",
+            "SAUCIE_embed_HVG_scaled",
+            "SAUCIE_gene_HVG_scaled"
+        ),
+        ATAC = c(
+            "Harmony_embed",
+            "scANVI*_embed",
+            "LIGER_embed",
+            "scVI_embed",
+            "ComBat_gene",
+            "scGen*_gene",
+            "Seurat v3 RPCA_gene",
+            "Seurat v3 CCA_gene",
+            "Scanorama_embed",
+            "MNN_gene",
+            "trVAE_embed",
+            "fastMNN_embed",
+            "BBKNN_graph",
+            "Scanorama_gene",
+            "fastMNN_gene",
+            "DESC_embed",
+            "SAUCIE_embed",
+            "SAUCIE_gene",
+            "Conos_graph"
+        )
+    )
```

Comment:
I should probably not hard code these but not sure if there is an easy way to get them. @martaint does/could your best methods functions output to best versions of each method? Would just need to be like what's here (method, output, features, scaling). Then I could make the best scatter functions more usable by somebody else.